docs/develop/01_sdk/02_cookbook/05_messaging/01_messaging.md

```diff
@@ -0,0 +1,53 @@
+---
+id: messaging_book
+title: Generate a Message
+---
+
+import SnippetBlock from '@site/src/components/SnippetBlock';
+
+import GenerateMessage from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/messaging/01_generate_message.ts';
+import EncryptMessage from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/messaging/02_encrypt_message.ts';
+import DecryptMessage from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/messaging/03_decrypt_message.ts';
+
+KILT uses a custom [Messaging protocol](../../../../concepts/06_messaging.md) that enables users of the KILT protocol to communicate via [unicast network](https://en.wikipedia.org/wiki/Unicast).
+The message types are defined by the actions that will be resolved by the receiver.
+Each of the messages sent are encrypted using the DID key agreement keys (see DID keys).
+The keys of both receiver and sender must be verifiable by querying the given DID of both parties.
+
+A message is formed from a message body that is a specific type along with the DID's URI senders and receivers.
+There are several types of structure for the demo will be generating a `request-credential` type.
+The message structure is checked and validated on the protocol level to ensure the users are sending correctly structured messages.
+
+<SnippetBlock
+  className="language-ts"
+>
+  {GenerateMessage}
+</SnippetBlock>
```

Comment:
the code examples look broken when they are rendered. I would rather use a `TsJsBlock` so that you can also see the imports.

Also the code doesn't get mentioned in the text. If we have a code example we should somehow mention it in the text? What does it do? What do we show? etc...

Reply:
I have included some example information.